pkg/vault/test/authorizer/TimelockAuthorizerDelays.test.ts

```diff
@@ -0,0 +1,372 @@
+        expectEvent.inReceipt(await receipt.wait(), 'ActionDelaySet', { actionId: ACTION_1, delay: ACTION_DELAY });
+      });
+    }
+
+    context('when the delay is being increased', () => {
+      // When incrasing the delay, the execution delay should always be the MINIMUM_EXECUTION_DELAY.
+
+      context('when there was no previous delay', () => {
+        itSchedulesTheDelayChangeCorrectly(MINIMUM_EXECUTION_DELAY);
+      });
+
+      context('when there was a previous delay set', () => {
+        sharedBeforeEach('set a previous smaller delay', async () => {
+          await authorizer.scheduleAndExecuteDelayChange(ACTION_1, ACTION_DELAY / 2, { from: root });
+        });
+
+        itSchedulesTheDelayChangeCorrectly(MINIMUM_EXECUTION_DELAY);
+      });
+    });
+
+    context('when the delay is being decreased', () => {
+      // When the delay is decreased, the execution delay should be the larger of the delay difference and
+      // MINIMUM_EXECUTION_DELAY.
+
+      context('when the previous delay was close to the new one', () => {
+        const previousDelay = ACTION_DELAY + DAY;
+
+        sharedBeforeEach(async () => {
+          await authorizer.scheduleAndExecuteDelayChange(ACTION_1, previousDelay, { from: root });
+        });
+
+        itSchedulesTheDelayChangeCorrectly(MINIMUM_EXECUTION_DELAY);
+      });
+
+      context('when the previous delay was much larger than the new one', () => {
+        const previousDelay = ACTION_DELAY + MONTH;
+
+        sharedBeforeEach(async () => {
+          await authorizer.scheduleAndExecuteDelayChange(ACTION_1, previousDelay, { from: root });
+        });
+
+        itSchedulesTheDelayChangeCorrectly(previousDelay - ACTION_DELAY);
+      });
+    });
+
+    describe('error scenarios', () => {
+      it('reverts if the sender is not root', async () => {
+        await expect(authorizer.scheduleDelayChange(ACTION_1, DAY, [], { from: other })).to.be.revertedWith(
+          'SENDER_IS_NOT_ROOT'
+        );
+      });
+
+      it('reverts if the new delay is more than 2 years', async () => {
+        await expect(
+          authorizer.scheduleDelayChange(ACTION_1, DAY * 365 * 2 + 1, [], { from: root })
+        ).to.be.revertedWith('DELAY_TOO_LARGE');
+      });
+
+      it('reverts if setDelay is called directly', async () => {
+        await expect(authorizer.instance.setDelay(ACTION_1, DAY)).to.be.revertedWith('CAN_ONLY_BE_SCHEDULED');
+      });
+
+      it('reverts if the delay is greater than the setAuthorizer delay', async () => {
+        const setAuthorizerDelay = await authorizer.delay(await actionId(vault, 'setAuthorizer'));
+        const id = await authorizer.scheduleDelayChange(ACTION_1, setAuthorizerDelay.add(1), [], { from: root });
+
+        // This condition is only tested at the time the delay is actually set (in case e.g. there was a scheduled action
+        // to change setAuthorizer's delay), so we must attempt to execute the action to get the expected revert.
```

Comment:
This is interesting.
We could also catch this at `scheduleDelayChange` time right?

Reply:
Potentially, though it would have some side effects. E.g. perhaps the new delay is not valid at the time of scheduling, but it will be valid at execution. E.g. during system setup, you'll need to first schedule and execute a setAuthroizer delay change, and only then will you be able to execute delay changes for other stuff.